File: internal/model/analytics.go
package model

// AnalyticsQuery holds the tunable window/limit for an analytics request.
type AnalyticsQuery struct {
	Days  int
	Limit int
}

// DailyPoint is one calendar day of the trend (zero-filled for empty days).
type DailyPoint struct {
	Date       string  `json:"date"` // YYYY-MM-DD
	Views      int     `json:"views"`
	WatchHours float64 `json:"watch_hours"`
}

// TopVideo is one row of the most-watched leaderboard within the window.
type TopVideo struct {
	VideoID      string  `json:"video_id"`
	Title        string  `json:"title"`
	ThumbnailKey string  `json:"thumbnail_key"`
	Views        int     `json:"views"`
	WatchHours   float64 `json:"watch_hours"`
}

// TopTag is one tag's view count within the window. Tag IDs are integers.
type TopTag struct {
	TagID int    `json:"tag_id"`
	Name  string `json:"name"`
	Views int    `json:"views"`
}

// AnalyticsSummary is the full payload returned by GET /api/admin/analytics.
type AnalyticsSummary struct {
	RangeDays         int          `json:"range_days"`
	TotalViews        int          `json:"total_views"`
	TotalWatchHours   float64      `json:"total_watch_hours"`
	AvgCompletionRate float64      `json:"avg_completion_rate"`
	ActiveUsers       int          `json:"active_users"`
	DailyTrend        []DailyPoint `json:"daily_trend"`
	TopVideos         []TopVideo   `json:"top_videos"`
	TopTags           []TopTag     `json:"top_tags"`
}

// DailyRawRow is one present day from the DB before zero-fill (seconds, not hours).
type DailyRawRow struct {
	Views          int
	WatchedSeconds int64
}

File: web/src/api/analytics.ts
import client from './client'

export interface DailyPoint {
  date: string
  views: number
  watch_hours: number
}
export interface TopVideo {
  video_id: string
  title: string
  thumbnail_key: string
  views: number
  watch_hours: number
}
export interface TopTag {
  tag_id: number
  name: string
  views: number
}
export interface AnalyticsSummary {
  range_days: number
  total_views: number
  total_watch_hours: number
  avg_completion_rate: number
  active_users: number
  daily_trend: DailyPoint[]
  top_videos: TopVideo[]
  top_tags: TopTag[]
}

export async function getAnalytics(days: number, limit = 10): Promise<AnalyticsSummary> {
  const res = await client.get<AnalyticsSummary>('/admin/analytics', { params: { days, limit } })
  return res.data
}
